compiler/src/iree/compiler/Codegen/LLVMGPU/Passes.cpp

```diff
@@ -271,18 +271,27 @@ static FailureOr<Value> gpuRequireMemSpaceAllocationFn(OpBuilder &builder,
                                                        MemRefType memRefType,
                                                        ValueRange dynamicSizes,
                                                        unsigned alignment) {
-  // Bail out if the memref type does not specify a memory space.
-  if (!llvm::isa_and_nonnull<gpu::AddressSpaceAttr>(
-          memRefType.getMemorySpace())) {
+  Attribute memorySpace = memRefType.getMemorySpace();
+  // Bail out if the memref type does not specify a GPU memory space.
+  if (memorySpace && !llvm::isa<gpu::AddressSpaceAttr>(memorySpace)) {
```

Comment:
```suggestion
  if (!isa_and_present<gpu::AddressSpaceAttr>(memorySpace)) {
```

Reply:
I actually need `(!isa)_and_present` rather than `!(isa_and_present)` so I switched back to what I had before.